### DISCUSSION TITLE
How does OpenFunction perform?

### DISCUSSION THREAD

**Discussion by ypjin** [OpenFunction/OpenFunction#343] [General]
Hi, wondering if there are any benchmarks for the performance of OpenFunction regarding cold start time, scaling time, etc. Since the sync runtime is based on Knative serving it may mostly rely on Knative, right? How about the async runtime? It should be OpenFunction specific. Nowadays the big players (AWS, Tencent, Alibaba, etc.) claim ms-level scaling speed. Just curious how far a self-hosted solution can go. 

Appreciate your response!
Best regards,

**Comment by benjaminhuo** [comment]
What you're asking is very important.
Currently, the async runtime scaling performance is like K8s HPA. And we're planning to improve it with the pod pool method which we've discussed in another thread

**Comment by ypjin** [comment]
Sounds great! Thank you!
